R/bundle.R

```diff
@@ -8,6 +8,13 @@ bundleAppDir <- function(appDir, appFiles) {
   for (file in appFiles) {
     from <- file.path(appDir, file)
     to <- file.path(bundleDir, file)
+    # if deploying a single-file Shiny application, name it "app.R" so it can
+    # be run as an ordinary Shiny application
+    if (!is.null(appPrimaryDoc) &&
+        tolower(tools::file_ext(appPrimaryDoc)) == "r" &&
+        file == appPrimaryDoc) {
+      to <- file.path(bundleDir, "app.R")
```

Comment:
That's a good question. It might be if the user ever knew we did it, but the rename only happens in the bundle, which the user never sees directly. Unless they have code that references the file by name (which seems unlikely given the generally self-contained nature of these apps) it's going to be seamless.